### DISCUSSION TITLE
Enhance `maximize-column`

### DISCUSSION THREAD

**Discussion by Paul-Eau** [YaLTeR/niri#820] [Ideas]
When using the mod+F shortcut on a window that doesn’t occupy the full width and height of the screen, only the width is maximized, not the height.

If a window is nested within a column, it would make sense for the first use of the shortcut to expel the window and make it single column / full height. If the shortcut is triggered again, the window should re-consume into its initial state.

**Comment by YaLTeR** [comment]
It intentionally only affects the column width, and in a column with multiple windows it is intentional that the heights stay as they are as the width is maximized. (It's called maximize-column and not maximize-window after all.)

Perhaps maximizing height makes sense in case it's a single window by itself which had been resized to less than the full monitor height, but the plan is to forbid doing this when floating windows are implemented, so this won't be a problem.

Something close enough to expelling and maximizing is fullscreening a window. It currently doesn't return it to the column upon unfullscreen, but I have an issue for this: #426.

**Comment by Paul-Eau** [comment]
Yes, of course, you're right. Just like maximize-column, could maximize-height be added then ? However, do you think it would be possible to see a maximize-window option one day ?

**Reply by YaLTeR** [reply]
There's reset-window-height that does this for individual windows

**Reply by Paul-Eau** [reply]
Thank you I hadn't thought of that !

**Comment by YaLTeR** [comment]
I'm implementing proper Wayland maximize here that removes gaps, borders and struts: https://github.com/YaLTeR/niri/pull/2376
